src/sentry/attachments/base.py

```diff
@@ -25,60 +48,112 @@ def from_upload(cls, file, **kwargs):
 
     @property
     def data(self):
-        if self._data is None and self._load is not None:
-            self._data = self._load()
+        if self._data is None and self._cache is not None:
+            assert self.id is not None
+
+            if self.chunks is None:
+                self._data = self._cache.get_unchunked_data(key=self.key, id=self.id)
+            else:
+                data = []
+                for chunk_index in range(self.chunks):
+                    data.append(
+                        self._cache.get_chunk(key=self.key, id=self.id, chunk_index=chunk_index)
+                    )
+
+                self._data = b"".join(data)
 
         return self._data
 
     def meta(self):
-        return {"name": self.name, "content_type": self.content_type, "type": self.type}
+        return prune_empty_keys(
+            {
+                "id": self.id,
+                "name": self.name,
+                "content_type": self.content_type,
+                "type": self.type,
+                "chunks": self.chunks,
+            }
+        )
 
 
 class BaseAttachmentCache(object):
-    def __init__(self, inner, appendix=None):
-        if appendix is None:
-            appendix = "a"
-        self.appendix = appendix
+    def __init__(self, inner):
         self.inner = inner
 
-    def make_key(self, key):
-        return u"{}:{}".format(key, self.appendix)
-
     def set(self, key, attachments, timeout=None):
-        key = self.make_key(key)
-        for index, attachment in enumerate(attachments):
-            compressed = zlib.compress(attachment.data)
-            self.inner.set(u"{}:{}".format(key, index), compressed, timeout, raw=True)
+        for id, attachment in enumerate(attachments):
+            if attachment.chunks is not None:
+                continue
+            # TODO(markus): We need to get away from sequential IDs, they
+            # are risking collision when using Relay.
+            if attachment.id is None:
+                attachment.id = id
```

Comment:
Agreed, let's fix this in python ingest in a follow-up PR, then remove this fallback.